### PR TITLE
code changes for registry api integration

### PR DESCRIPTION
**Pull request by vrushalijoshi**

### CODE REVIEW

**Review by rtay1188 — APPROVED** [review]
Looks good, only thing is possibly use https://golang.cafe/blog/golang-functional-options-pattern.html for initializing registry client

File: go.mod [hunk 1]
```diff
@@ -3,9 +3,13 @@ module github.com/istio-ecosystem/admiral-sharding-manager
 go 1.22.4
 
 require (
-	github.com/istio-ecosystem/admiral-api v1.0.2
+	github.com/google/go-cmp v0.6.0
+	github.com/google/uuid v1.6.0
+	github.com/istio-ecosystem/admiral-api v1.0.3-0.20240604180405-9243983d25e2
```
Comment (nirvanagit) [inline go.mod]: Created a new release for admiral-api https://github.com/istio-ecosystem/admiral-api/releases/tag/v1.1.0

File: pkg/testdata/error-test-shard-identity-bulk.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (nirvanagit) [inline pkg/testdata/error-test-shard-identity-bulk.json:1]: What's the error in this file?

File: pkg/testdata/error-test-shard-identity.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (nirvanagit) [inline pkg/testdata/error-test-shard-identity.json:1]: What's the error in this file?

File: pkg/testdata/test-shard-identity-bulk.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
+  "clusters": [
+    {
+      "name": "cluster1",
+      "locality": "us-west-2",
+      "metadata": {
+        "segment": "seg"
```
Comment (nirvanagit) [inline pkg/testdata/test-shard-identity-bulk.json:7]: What is segment?

File: pkg/testdata/test-cluster-identity.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (nirvanagit) [inline pkg/testdata/test-cluster-identity.json:1]: Is this the data returned for a given shard (e.g. a business unit) ? 
If yes, then can we rename the test file to be something like: `test-cluster-identity-per-shard`

File: pkg/testdata/test-cluster-identity.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+  "clustername": "cluster1",
+  "assetMetadata": [
```
Comment (nirvanagit) [inline pkg/testdata/test-cluster-identity.json]: I know we decided with `assetMetadata` in the design, but rethinking/reevaluating. May be `assetList` is better property name.

File: go.mod (hunk 1 above)
Comment (vrushalijoshi) [reply go.mod]: thanks @nirvanagit , updated the version

File: pkg/testdata/error-test-shard-identity-bulk.json (hunk 2 above)
Comment (vrushalijoshi) [reply pkg/testdata/error-test-shard-identity-bulk.json:1]: In this configured "assets": "test-value" is string but expected is array

File: pkg/testdata/error-test-shard-identity.json (hunk 3 above)
Comment (vrushalijoshi) [reply pkg/testdata/error-test-shard-identity.json:1]: In this one "metadata": "test-value" is set as string but expected with IdentityMetdata type

File: pkg/testdata/test-shard-identity-bulk.json (hunk 4 above)
Comment (vrushalijoshi) [reply pkg/testdata/test-shard-identity-bulk.json:7]: I just added a place holder metadata value, currently not used

File: pkg/testdata/test-cluster-identity.json (hunk 5 above)
Comment (vrushalijoshi) [reply pkg/testdata/test-cluster-identity.json:1]: no, this is the data for all the identities for a given cluster name

**Review by nirvanagit — DISMISSED** [review]

**Comment by nirvanagit** [comment]
Ensure the DCO passes before merging.

File: pkg/common/common.go [hunk 7]
```diff
@@ -0,0 +1 @@
+package common
```
Comment (nirvanagit) [inline pkg/common/common.go]: Can we rename this to util.

File: pkg/testdata/test-cluster-identity.json (hunk 6 above)
Comment (vrushalijoshi) [reply pkg/testdata/test-cluster-identity.json]: updated to use assetList

**Review by nirvanagit — APPROVED** [review]

**Review by rtay1188 — APPROVED** [review]

**Review by nirvanagit — APPROVED** [review]

**Review by rtay1188 — APPROVED** [review]

**Pull request merged**